receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/docker_stats.go

```diff
@@ -23,12 +23,12 @@ type ContainerStats struct {
 	Read         time.Time `json:"read"`
 	PreviousRead time.Time `json:"preread"`
 
-	Memory      MemoryStats             `json:"memory_stats,omitempty"`
-	Disk        DiskStats               `json:"blkio_stats,omitempty"`
+	Memory      *MemoryStats            `json:"memory_stats,omitempty"`
```

Comment:
I think generally struct works fine when serializing (as opposed to parsing where emptyness information goes away) - what's the motivation for the pointer changes?

Reply:
I think using pointer makes it easier to check for missing struct (null) rather than doing an empty struct check ( `someObject == myStruct{}`). Maybe I am not aware of any core/performance difference of pointer vs struct. 

Is there any major disadvantage if we use pointer here? Can you please enlighten me? If so, I will take it back and push another update with general struct.

Reply:
Sorry - I realized these probably are actually parsed, not used for serialization. So pointer seems fine